src/components/decimal/decimal.component.js

```diff
@@ -11,247 +16,261 @@ import LocaleContext from "../../__internal__/i18n-context";
-    const value = isControlled
-      ? this.getSafeValueProp(true)
-      : this.props.defaultValue || this.emptyValue;
-
-    this.state = {
-      value,
-      visibleValue: this.formatValue(value),
-      isControlled,
-    };
-  }
-
-  componentDidUpdate(prevProps, prevState) {
-    const message =
-      "Input elements should not switch from uncontrolled to controlled (or vice versa). " +
-      "Decide between using a controlled or uncontrolled input element for the lifetime of the component";
-    const isControlled = this.isControlled();
-    invariant(this.state.isControlled === isControlled, message);
-    if (prevProps.precision !== this.props.precision) {
-      // eslint-disable-next-line no-console
-      console.error(
-        "Decimal `precision` prop has changed value. Changing the Decimal `precision` prop has no effect."
-      );
-    }
-    if (isControlled) {
-      const valueProp = this.getSafeValueProp();
-      if (valueProp !== prevState.value) {
-        this.setState({
-          value: valueProp,
-          visibleValue: this.formatValue(valueProp),
-        });
-      }
-    }
-  }
-
-  callOnChange = () => {
-    if (this.props.onChange) {
-      this.props.onChange(this.createEvent());
-    }
-  };
-
-  onChange = (ev) => {
-    const {
-      target: { value },
-    } = ev;
-    this.setState(
-      { value: this.toStandardDecimal(value), visibleValue: value },
-      () => {
-        this.callOnChange();
-      }
-    );
-  };
-
-  onBlur = () => {
-    let shouldCallOnChange = false;
-    this.setState(
-      ({ value, visibleValue }) => {
-        if (!visibleValue) {
-          shouldCallOnChange = value !== this.emptyValue;
-          return {
-            value: this.emptyValue,
-            visibleValue: this.formatValue(this.emptyValue),
-          };
-        }
-        return {
-          visibleValue: this.formatValue(value),
-        };
-      },
-      () => {
-        if (shouldCallOnChange) {
-          this.callOnChange();
-        }
-        if (this.props.onBlur) {
-          this.props.onBlur(this.createEvent());
-        }
-      }
-    );
-  };
-
-  createEvent = () => {
-    const standardVisible = this.toStandardDecimal(this.state.visibleValue);
-    const formattedValue = this.isNaN(standardVisible)
-      ? this.state.visibleValue
-      : this.formatValue(standardVisible);
-    return {
-      target: {
-        name: this.props.name,
-        id: this.props.id,
-        value: {
-          rawValue: this.state.value,
-          formattedValue,
-        },
-      },
-    };
-  };
-
-  /**
-   * Determine if the component is controlled at the time of call
-   */
-  isControlled() {
-    return this.props.value !== undefined;
-  }
-
-  isNaN = (value) => {
-    return Number.isNaN(Number(value));
-  };
-
-  getSafeValueProp = (isInitialValue) => {
-    const { value, allowEmptyValue } = this.props;
+  const getSafeValueProp = (isInitialValue) => {
```

Comment:
To me `isInitialValue` suggests a boolean, maybe calling it `initialValue` would be a bit more intuitive?